src/Database/Esqueleto/PostgreSQL.hs

```diff
@@ -205,17 +206,25 @@ upsertBy
     -- ^ the record in the database after the operation
 upsertBy uniqueKey record updates = do
     sqlB <- R.ask
-    maybe
-      (throw (UnexpectedCaseErr OperationNotSupported)) -- Postgres backend should have connUpsertSql, if this error is thrown, check changes on persistent
-      (handler sqlB)
-      (connUpsertSql sqlB)
+    case connUpsertSql sqlB of
+        Nothing ->
+            -- Postgres backend should have connUpsertSql, if this error is
+            -- thrown, check changes on persistent
+            throw (UnexpectedCaseErr OperationNotSupported)
+        Just upsertSql -> do
+            let (updateText, updateVals) =
+                    updatesText sqlB
+                queryText =
+                    upsertSql entDef uniqueFields updateText
+                queryVals =
+                    addVals updateVals
+            xs <- rawSql queryText queryVals
+            pure (head xs)
```

Comment:
`connUpsertSql` was changed to take the field names directly. This is the only semantic change in the PR.